File: src/backend/spring-boot-club-manager/src/main/java/com/archeryclub/springbootclubmanager/entity/Member.java
package com.archeryclub.springbootclubmanager.entity;

import lombok.Data;
import jakarta.persistence.*;

@Entity
@Table(name="members")
@Data
public class Member {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @Column(name = "firstName")
    private String firstName;

    @Column(name = "lastName")
    private String lastName;

    @Column(name = "studentId")
    private Long studentId;

    @Column(name = "uciEmail")
    private String uciEmail;

    @Column(name = "active")
    private boolean active;

    @Column(name = "loaningEquipment")
    private boolean loaningEquipment;
}

File: src/frontend/typings/Member.ts
class Member {
  id?: number;
  studentId: string;
  firstName: string;
  lastName: string;
  email: string;
  active?: boolean;
  owesEquipment: boolean;

  constructor(
    id: number,
    studentId: string,
    lastName: string,
    firstName: string,
    email: string,
    active: boolean,
    owesEquipment: boolean
  ) {
    this.id = id;
    this.studentId = studentId;
    this.firstName = firstName;
    this.lastName = lastName;
    this.email = email;
    this.active = active;
    this.owesEquipment = owesEquipment;
  }
}
